extension/src/pipeline/util.test.ts

```diff
@@ -0,0 +1,237 @@
+    )
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      2,
+      ['actual', 'prob'],
+      'Pick a Metric for X'
+    )
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      3,
+      ['prob'],
+      'Pick a Metric for Y'
+    )
+    expect(result).toStrictEqual({
+      dataFile: 'file.json',
+      template: 'simple',
+      x: 'actual',
+      y: 'prob'
+    })
+  })
+
+  it('should return early if user does not pick a template', async () => {
+    mockedPickFile.mockResolvedValueOnce('file.json')
+    mockedLoadJson.mockReturnValueOnce(mockValidData)
+    mockedQuickPickOne.mockResolvedValueOnce(undefined)
+
+    const result = await pickPlotConfiguration()
+
+    expect(mockedQuickPickOne).toHaveBeenCalledTimes(1)
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      1,
+      [
+        'simple',
+        'linear',
+        'confusion',
+        'confusion_normalized',
+        'scatter',
+        'scatter_jitter',
+        'smooth',
+        'bar_horizontal_sorted',
+        'bar_horizontal'
+      ],
+      'Pick a Plot Template'
+    )
+
+    expect(result).toStrictEqual(undefined)
+  })
+
+  it('should return early if user does not pick a x field', async () => {
+    mockedPickFile.mockResolvedValueOnce('file.json')
+    mockedLoadJson.mockReturnValueOnce(mockValidData)
+    mockedQuickPickOne
+      .mockResolvedValueOnce('simple')
+      .mockResolvedValueOnce(undefined)
+
+    const result = await pickPlotConfiguration()
+
+    expect(mockedQuickPickOne).toHaveBeenCalledTimes(2)
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      1,
+      [
+        'simple',
+        'linear',
+        'confusion',
+        'confusion_normalized',
+        'scatter',
+        'scatter_jitter',
+        'smooth',
+        'bar_horizontal_sorted',
+        'bar_horizontal'
+      ],
+      'Pick a Plot Template'
+    )
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      2,
+      ['actual', 'prob'],
+      'Pick a Metric for X'
+    )
+    expect(result).toStrictEqual(undefined)
+  })
+
+  it('should return early if user does not pick a y field', async () => {
+    mockedPickFile.mockResolvedValueOnce('file.json')
+    mockedLoadJson.mockReturnValueOnce(mockValidData)
+    mockedQuickPickOne
+      .mockResolvedValueOnce('simple')
+      .mockResolvedValueOnce('actual')
+      .mockResolvedValueOnce(undefined)
+
+    const result = await pickPlotConfiguration()
+
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      1,
+      [
+        'simple',
+        'linear',
+        'confusion',
+        'confusion_normalized',
+        'scatter',
+        'scatter_jitter',
+        'smooth',
+        'bar_horizontal_sorted',
+        'bar_horizontal'
+      ],
+      'Pick a Plot Template'
+    )
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
+      2,
+      ['actual', 'prob'],
+      'Pick a Metric for X'
+    )
+    expect(mockedQuickPickOne).toHaveBeenNthCalledWith(
```

Comment:
[Q] Do we need these `toHaveBeenNthCalledWith` assertions in the early return tests? Could we just have a `toHaveBeenCalledTimes` and avoid the duplication?